sklearn/impute/_knn.py

```diff
@@ -55,9 +55,9 @@ class KNNImputer(_BaseImputer):
 
         - 'nan_euclidean'
         - callable : a user-defined function which conforms to the definition
-          of ``_pairwise_callable(X, Y, metric, **kwds)``. The function
-          accepts two arrays, X and Y, and a `missing_values` keyword in
-          `kwds` and returns a scalar distance value.
+          of ``_pairwise_callable(X, Y, *, missing_values=np.nan)``. The function
+          accepts two 1-D arrays, X and Y, and a `missing_values` keyword and 
+          returns a scalar distance value.
```

Comment:
I would slightly change the formulation to use `x` and `y` to acknowledge that we have 1-D array.

```suggestion
        - callable : a user-defined function which conforms to the definition
          of ``func_metric(x, y, *, missing_values=np.nan)``. `x` and `y`
          corresponds to a row (i.e. 1-D arrays) of `X` and `Y`, respectively.
          The callable should returns a scalar distance value.
```